### DISCUSSION TITLE
Condition on token and topics

### DISCUSSION THREAD

**Discussion by Jcs1994** [kreait/firebase-php#582] [Q&A] [answered]
Hello,

Is it possible to set the target using multiple conditions on token **AND** topics

For example :
```
$my_token = "MyToken";

$message = CloudMessage::withTarget('condition', "token = '$token' && TopicA in topics")
                ->withNotification(Notification::create(null, "test"));
```
When I send this message I get `Kreait\Firebase\Exception\Messaging\InvalidMessage; Message: Request contains an invalid argument.`

I also tried with `CloudMessage::fromArray()` using `token` and `topic` keys, but I get an error. 
=> `Message: A message can only have one of the following targets: condition, token, topic`

**Comment by jeromegamez** [comment, marked as answer]
Conditions only work with topics, conditions based on a token are not supported (at least as far as I know).

Based on your message, I assume that you want to send a message to a token, but only if it is subscribed to the `TopicA` topic. In that case you would have to check if the token is subscribed to the topic beforehand:

```php
if ($messaging->getAppInstance($token)->isSubscribedToTopic('TopicA')) {
    $message = CloudMessage::withTarget('token', $token)->withNotification(Notification::create(null, "test"));
    $messaging->send($message);
}
